Use newer version of microhttpd

In ArchLinux we are already using a newer version of microhttpd: `Name           : libmicrohttpd
Version        : 0.9.44-1`

With this version we are getting a deprecated function usage warning:
```
[ 66%] Built target exp
[ 67%] Building CXX object libweb3jsonrpc/CMakeFiles/web3jsonrpc.dir/JsonHelper.cpp.o
[ 69%] Building CXX object libweb3jsonrpc/CMakeFiles/web3jsonrpc.dir/WinPipeServer.cpp.o
[ 71%] Building CXX object libweb3jsonrpc/CMakeFiles/web3jsonrpc.dir/SafeHttpServer.cpp.o
[ 72%] Building CXX object libweb3jsonrpc/CMakeFiles/web3jsonrpc.dir/Bzz.cpp.o
/home/lefteris/ew/webthree/libweb3jsonrpc/SafeHttpServer.cpp:42:32: warning: 'MHD_create_response_from_data' is deprecated: MHD_create_response_from_data() is deprecated, use MHD_create_response_from_buffer() [-Wdeprecated-declarations]
        struct MHD_Response *result = MHD_create_response_from_data(_response.size(),(void *) _response.c_str(), 0, 1);
                                      ^
/usr/include/microhttpd.h:2022:1: note: 'MHD_create_response_from_data' has been explicitly marked deprecated here
MHD_create_response_from_data (size_t size,
^
/home/lefteris/ew/webthree/libweb3jsonrpc/SafeHttpServer.cpp:55:32: warning: 'MHD_create_response_from_data' is deprecated: MHD_create_response_from_data() is deprecated, use MHD_create_response_from_buffer() [-Wdeprecated-declarations]
        struct MHD_Response *result = MHD_create_response_from_data(0, NULL, 0, 1);
                                      ^
/usr/include/microhttpd.h:2022:1: note: 'MHD_create_response_from_data' has been explicitly marked deprecated here
MHD_create_response_from_data (size_t size,
^
2 warnings generated.
[ 74%] Building CXX object libweb3jsonrpc/CMakeFiles/web3jsonrpc.dir/Whisper.cpp.o
```

It would be quite nice if we switched to using the new API since at some point the deprecated function may just dissapear from the API.

hmm I get errors instead of warning here

```
/home/null/code/webthree-umbrella/webthree/libweb3jsonrpc/SafeHttpServer.cpp: In member function 'virtual bool dev::SafeHttpServer::SendResponse(const string&, void*)':
/home/null/code/webthree-umbrella/webthree/libweb3jsonrpc/SafeHttpServer.cpp:42:32: error: 'MHD_Response* MHD_create_response_from_data(size_t, void*, int, int)' is deprecated [-Werror=deprecated-declarations]
  struct MHD_Response *result = MHD_create_response_from_data(_response.size(),(void *) _response.c_str(), 0, 1);
                                ^
In file included from /home/null/code/webthree-umbrella/webthree/libweb3jsonrpc/SafeHttpServer.cpp:23:0:
/usr/include/microhttpd.h:2022:1: note: declared here
 MHD_create_response_from_data (size_t size,
 ^
/home/null/code/webthree-umbrella/webthree/libweb3jsonrpc/SafeHttpServer.cpp:42:32: error: 'MHD_Response* MHD_create_response_from_data(size_t, void*, int, int)' is deprecated [-Werror=deprecated-declarations]
  struct MHD_Response *result = MHD_create_response_from_data(_response.size(),(void *) _response.c_str(), 0, 1);
                                ^
In file included from /home/null/code/webthree-umbrella/webthree/libweb3jsonrpc/SafeHttpServer.cpp:23:0:
/usr/include/microhttpd.h:2022:1: note: declared here
 MHD_create_response_from_data (size_t size,
 ^
/home/null/code/webthree-umbrella/webthree/libweb3jsonrpc/SafeHttpServer.cpp:42:111: error: 'MHD_Response* MHD_create_response_from_data(size_t, void*, int, int)' is deprecated [-Werror=deprecated-declarations]
  struct MHD_Response *result = MHD_create_response_from_data(_response.size(),(void *) _response.c_str(), 0, 1);
                                                                                                               ^
In file included from /home/null/code/webthree-umbrella/webthree/libweb3jsonrpc/SafeHttpServer.cpp:23:0:
/usr/include/microhttpd.h:2022:1: note: declared here
 MHD_create_response_from_data (size_t size,
 ^
/home/null/code/webthree-umbrella/webthree/libweb3jsonrpc/SafeHttpServer.cpp: In member function 'virtual bool dev::SafeHttpServer::SendOptionsResponse(void*)':
/home/null/code/webthree-umbrella/webthree/libweb3jsonrpc/SafeHttpServer.cpp:55:32: error: 'MHD_Response* MHD_create_response_from_data(size_t, void*, int, int)' is deprecated [-Werror=deprecated-declarations]
  struct MHD_Response *result = MHD_create_response_from_data(0, NULL, 0, 1);
                                ^
In file included from /home/null/code/webthree-umbrella/webthree/libweb3jsonrpc/SafeHttpServer.cpp:23:0:
/usr/include/microhttpd.h:2022:1: note: declared here
 MHD_create_response_from_data (size_t size,
 ^
/home/null/code/webthree-umbrella/webthree/libweb3jsonrpc/SafeHttpServer.cpp:55:32: error: 'MHD_Response* MHD_create_response_from_data(size_t, void*, int, int)' is deprecated [-Werror=deprecated-declarations]
  struct MHD_Response *result = MHD_create_response_from_data(0, NULL, 0, 1);
                                ^
In file included from /home/null/code/webthree-umbrella/webthree/libweb3jsonrpc/SafeHttpServer.cpp:23:0:
/usr/include/microhttpd.h:2022:1: note: declared here
 MHD_create_response_from_data (size_t size,
 ^
/home/null/code/webthree-umbrella/webthree/libweb3jsonrpc/SafeHttpServer.cpp:55:75: error: 'MHD_Response* MHD_create_response_from_data(size_t, void*, int, int)' is deprecated [-Werror=deprecated-declarations]
  struct MHD_Response *result = MHD_create_response_from_data(0, NULL, 0, 1);
                                                                           ^
In file included from /home/null/code/webthree-umbrella/webthree/libweb3jsonrpc/SafeHttpServer.cpp:23:0:
/usr/include/microhttpd.h:2022:1: note: declared here
 MHD_create_response_from_data (size_t size,
 ^
cc1plus: all warnings being treated as errors
webthree/libweb3jsonrpc/CMakeFiles/web3jsonrpc.dir/build.make:302: recipe for target 'webthree/libweb3jsonrpc/CMakeFiles/web3jsonrpc.dir/SafeHttpServer.cpp.o' failed
```

--- a/libweb3jsonrpc/SafeHttpServer.cpp
+++ b/libweb3jsonrpc/SafeHttpServer.cpp
@@ -39,7 +39,11 @@ struct mhd_coninfo
 bool SafeHttpServer::SendResponse(string const& _response, void* _addInfo)
 {
 	struct mhd_coninfo* client_connection = static_cast<struct mhd_coninfo*>(_addInfo);
-	struct MHD_Response *result = MHD_create_response_from_data(_response.size(),(void *) _response.c_str(), 0, 1);
+	struct MHD_Response *result = MHD_create_response_from_buffer(
+		_response.size(),
+		static_cast<void *>(const_cast<char *>(_response.c_str())),
+		MHD_RESPMEM_MUST_COPY
+	);
 
 	MHD_add_response_header(result, "Content-Type", "application/json");
 	MHD_add_response_header(result, "Access-Control-Allow-Origin", m_allowedOrigin.c_str());
@@ -52,7 +56,7 @@ bool SafeHttpServer::SendResponse(string const& _response, void* _addInfo)
 bool SafeHttpServer::SendOptionsResponse(void* _addInfo)
 {
 	struct mhd_coninfo* client_connection = static_cast<struct mhd_coninfo*>(_addInfo);
-	struct MHD_Response *result = MHD_create_response_from_data(0, NULL, 0, 1);
+	struct MHD_Response *result = MHD_create_response_from_buffer(0, nullptr, MHD_RESPMEM_MUST_COPY);
 
 	MHD_add_response_header(result, "Allow", "POST, OPTIONS");
 	MHD_add_response_header(result, "Access-Control-Allow-Origin", m_allowedOrigin.c_str());

